### PR TITLE
Add --log arg to enable multiple log targets

### PR DESCRIPTION
**Pull request by adonagy**

### CODE REVIEW

File: light_node/src/configuration.rs [hunk 1]
```diff
@@ -1089,4 +1201,45 @@ impl Environment {
...
         }
     }
+
+    pub fn create_logger(&self) -> Logger {
+        let Environment { logging, .. } = self;
+
+        let log_file = match &logging.file {
+            Some(file) => file.clone(),
+            None => {
+                let mut path = PathBuf::new();
```
Comment (bkontur) [inline light_node/src/configuration.rs:1211]: @adonagy 
try this `PathBuf::from`

File: light_node/src/configuration.rs [hunk 2]
```diff
@@ -1089,4 +1201,45 @@ impl Environment {
             validate_cfg_identity_and_stop: args.is_present("validate-cfg-identity-and-stop"),
         }
     }
+
+    pub fn create_logger(&self) -> Logger {
+        let Environment { logging, .. } = self;
+
+        let log_file = match &logging.file {
```
Comment (bkontur) [inline light_node/src/configuration.rs:1208]: @adonagy 
I would handle file only if LoggerType::FileLogger is choosen

File: light_node/src/configuration.rs [hunk 3]
```diff
@@ -1089,4 +1201,45 @@ impl Environment {
...
+            Some(file) => file.clone(),
+            None => {
+                let mut path = PathBuf::new();
+                path.push("./tezedge.log");
+                path
+            }
+        };
+
+        if logging.log.contains(&LoggerType::FileLogger)
```
[the remark this answers is not in the document]
Comment (adonagy) [reply light_node/src/configuration.rs:1217]: Yeah, you are right, this solution is messy. I'll try another approach

[the remark this answers is not in the document]
Comment (adonagy) [reply light_node/src/configuration.rs]: @bkontur I've added a third variant to the enum. This approach is way cleaner and don't introduce unnecessary multiple values for the argument.

File: light_node/src/configuration.rs [hunk 4]
```diff
@@ -1089,4 +1203,50 @@ impl Environment {
...
+            Logger::root(
+                drains[0].clone().filter_level(logging.level).fuse(),
+                slog::o!(),
+            )
+        } else {
+            // combine 2 or more drains into Duplicates
+
+            // need an initial value for fold, create it from the first two drains in the vector
+            let initial_value =
+                Box::new(Duplicate::new(drains[0].clone(), drains[1].clone()).fuse());
```
Comment (bkontur) [inline light_node/src/configuration.rs:1236]: @adonagy 
this will failes if drains.is_empty

Comment (adonagy) [reply light_node/src/configuration.rs:1243]: @bkontur  Fixed

File: light_node/src/main.rs [hunk 5]
```diff
@@ -538,8 +479,11 @@ fn main() {
             )
         });
 
-    // Creates default logger
-    let log = create_logger(&env);
+    // Creates loggers
+    let log = match env.create_logger() {
+        Ok(log) => log,
+        Err(e) => panic!("Error while creating loggers: {:?}", e),
```
Comment (bkontur) [inline light_node/src/main.rs:485]: @adonagy 
```
panic!("Error while creating loggers, check '--log' argument, reason: {:?}", e),
```

**Review by bkontur — APPROVED** [review]
@adonagy
please, change the panic error,
and squash, rebase, merge to develop

**Pull request merged**